internal/config/config.go

```diff
@@ -234,26 +238,35 @@ func NewConfig(attrs Attributes) (*Config, error) {
 	return cfg, nil
 }
 
+func getFlagNameFromProfile(oktaProfile string, flag string) string {
+	if oktaProfile == "" {
+		return flag
+	}
+	return fmt.Sprintf("%s.%s", oktaProfile, flag)
+}
+
 func readConfig() (Attributes, error) {
+	oktaProfile := viper.GetString(OktaProfileFlag)
+
 	attrs := Attributes{
-		AWSCredentials:      viper.GetString(AWSCredentialsFlag),
-		AWSIAMIdP:           viper.GetString(AWSIAMIdPFlag),
-		AWSIAMRole:          viper.GetString(AWSIAMRoleFlag),
-		AWSSessionDuration:  viper.GetInt64(SessionDurationFlag),
-		Debug:               viper.GetBool(DebugFlag),
-		DebugAPICalls:       viper.GetBool(DebugAPICallsFlag),
-		DebugConfig:         viper.GetBool(DebugConfigFlag),
-		FedAppID:            viper.GetString(AWSAcctFedAppIDFlag),
-		Format:              viper.GetString(FormatFlag),
-		LegacyAWSVariables:  viper.GetBool(LegacyAWSVariablesFlag),
-		ExpiryAWSVariables:  viper.GetBool(ExpiryAWSVariablesFlag),
-		CacheAccessToken:    viper.GetBool(CacheAccessTokenFlag),
-		OIDCAppID:           viper.GetString(OIDCClientIDFlag),
-		OpenBrowser:         viper.GetBool(OpenBrowserFlag),
-		OrgDomain:           viper.GetString(OrgDomainFlag),
-		Profile:             viper.GetString(ProfileFlag),
-		QRCode:              viper.GetBool(QRCodeFlag),
-		WriteAWSCredentials: viper.GetBool(WriteAWSCredentialsFlag),
+		AWSCredentials:      viper.GetString(getFlagNameFromProfile(oktaProfile, AWSCredentialsFlag)),
```

Comment:
If the .env file is present and config is not, which config values would actually need to exist to work to use profiles? `profilename.OKTA_AWS_CLI_ORG_DOMAIN` ?